src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -0,0 +1,1171 @@
+	 *
+	 * Use this for cases where you are processing chunks of HTML that
+	 * will be found within a bigger HTML document, such as rendered
+	 * block output that exists within a post, `the_content` inside a
+	 * rendered site layout.
+	 *
+	 * Fragment parsing occurs within a context, which is an HTML element
+	 * that the document will eventually be placed in. It becomes important
+	 * when special elements have different rules than others, such as inside
+	 * a TEXTAREA or a TITLE tag where things that look like tags are text,
+	 * or inside a SCRIPT tag where things that look like HTML syntax are JS.
+	 *
+	 * The context value should be a representation of the tag into which the
+	 * HTML is found. For most cases this will be the body element. The HTML
+	 * form is provided because a context element may have attributes that
+	 * impact the parse, such as with a SCRIPT tag and its `type` attribute.
+	 *
+	 * ## Current HTML Support
+	 *
+	 *  - The only supported context is `<body>`, which is the default value.
+	 *  - The only supported document encoding is `UTF-8`, which is the default value.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $html     Input HTML fragment to process.
+	 * @param string $context  Context element for the fragment, must be default of `<body>`.
+	 * @param string $encoding Text encoding of the document; must be default of 'UTF-8'.
+	 * @return WP_HTML_Processor|null The created processor if successful, otherwise null.
+	 */
+	public static function createFragment( $html, $context = '<body>', $encoding = 'UTF-8' ) {
+		if ( '<body>' !== $context ) {
+			return null;
+		}
+
+		$p = new self( $html, self::CONSTRUCTOR_UNLOCK_CODE );
+		$p->state->context_node   = array( 'BODY', array() );
+		$p->state->insertion_mode = WP_HTML_Processor_State::INSERTION_MODE_IN_BODY;
+
+		// @TODO: Create "fake" bookmarks for noni-existent but implied nodes.
+		$p->bookmarks['root-node'] = new WP_HTML_Span( 0, 0 );
+		$p->bookmarks['context-node'] = new WP_HTML_Span( 0, 0 );
+
+		$p->state->stack_of_open_elements->push(
+			new WP_HTML_Token(
+				'root-node',
+				'HTML',
+				false
+			)
+		);
+
+		$p->state->stack_of_open_elements->push(
+			new WP_HTML_Token(
+				'context-node',
+				$p->state->context_node[0],
+				false
+			)
+		);
+
+		return $p;
+	}
+
+	/**
+	 * Constructor.
+	 *
+	 * @access private
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string      $html                                  HTML to process.
+	 * @param string|null $use_the_static_create_methods_instead This constructor should not be called manually.
+	 */
+	public function __construct( $html, $use_the_static_create_methods_instead = null ) {
+		parent::__construct( $html );
+
+		if ( self::CONSTRUCTOR_UNLOCK_CODE !== $use_the_static_create_methods_instead ) {
+			_doing_it_wrong(
+				__METHOD__,
+				'Call WP_HTML_Processor::createFragment to create an HTML Processor instead of calling the constructor directly.',
```

Comment:
```suggestion
				sprintf(
					/* translators: %s: WP_HTML_Processor::createFragment. */
					__( 'Call %s to create an HTML Processor instead of calling the constructor directly.' ),
					'<code>WP_HTML_Processor::createFragment</code>'
				)
```
The `$message` should be translatable with code references passed via `sprintf()` and wrapped in `<code>` tags.

🔢 Applies elsewhere in the PR.